packages/vue-components/lib/components/MapboxMap/MapboxMap.vue

```diff
@@ -43,30 +43,51 @@ export default {
     mapboxStyle: {
       type: String,
       default: MAPBOX_STYLE
-    }
+    },
+    padding: {
+      type: Object,
+      default: () => ({ left: 0, right: 0, top: 0, bottom: 0 }),
+      validator: value => {
+        return Object.prototype.hasOwnProperty.call(value, 'top') || 
+               Object.prototype.hasOwnProperty.call(value, 'right') || 
+               Object.prototype.hasOwnProperty.call(value, 'bottom') || 
+               Object.prototype.hasOwnProperty.call(value, 'left')
+      }
+    },
+  },
+  watch: {
+    padding: {
+      handler: function(newValue) {
+        this.$refs.vmapbox.map.setPadding(newValue)
+      },
+      deep: true,
```

Comment:
Just wondering 🤔 : If you make this `immediate`, would that negate the necessity of having to call `map.setPadding` in `mounted`? Or would it then be called too early? I even wonder, is the `map` always ready, since mapbox needs to be initialized and isn't instantly ready..

Reply:
The map was not ready when I used `immediate`, so that's where the call in `mounted` came from 😔

Reply:
clear!